File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.Object/Convert/ToValueType/Object.ToNullableDateTimeOffSet.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library


Public Module Extensions_393

	''' <summary>
	'''     An object extension method that converts the @this to a nullable date time off set.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>@this as a DateTimeOffset?</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function ToNullableDateTimeOffSet(this As Object) As System.Nullable(Of DateTimeOffset)
		If this Is Nothing OrElse this = DBNull.Value Then
			Return Nothing
		End If

		Return New DateTimeOffset(Convert.ToDateTime(this), TimeSpan.Zero)
	End Function
End Module

File: src/Z.Core/System.Object/Convert/ToValueType/Object.ToNullableDateTimeOffSet.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System;

public static partial class Extensions
{
    /// <summary>
    ///     An object extension method that converts the @this to a nullable date time off set.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>@this as a DateTimeOffset?</returns>
    public static DateTimeOffset? ToNullableDateTimeOffSet(this object @this)
    {
        if (@this == null || @this == DBNull.Value)
        {
            return null;
        }

        return new DateTimeOffset(Convert.ToDateTime(@this), TimeSpan.Zero);
    }
}